staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff.go

```diff
@@ -65,11 +65,11 @@ func (az *Cloud) Event(obj runtime.Object, eventtype, reason, message string) {
 }
 
 // GetVirtualMachineWithRetry invokes az.getVirtualMachine with exponential backoff retry
-func (az *Cloud) GetVirtualMachineWithRetry(name types.NodeName) (compute.VirtualMachine, error) {
+func (az *Cloud) GetVirtualMachineWithRetry(name types.NodeName, crt cacheReadType) (compute.VirtualMachine, error) {
 	var machine compute.VirtualMachine
 	var retryErr error
 	err := wait.ExponentialBackoff(az.RequestBackoff(), func() (bool, error) {
-		machine, retryErr = az.getVirtualMachine(name)
+		machine, retryErr = az.getVirtualMachine(name, crt)
```

Comment:
@feiskyer - do we still need the  `wait.ExponentialBackoff` even when the SDK does that on behalf?

Reply:
I have plans to remove the backoff logic from SDK, and let the cloud provider fully control it. We have seen a lot of issues because of SDK retries, e.g. infinite 429 retries.

Reply:
I think better off relying on SDK, not our code. the SDK can then adapt the RP interface as it changes. i.e. supporting retry-after etc.

Reply:
with the SDK, we couldn't get the retry-after header back. So it's hard to determine when could cloud-provider to send new requests again.

And also, the retry-after returned by ARM may be too long for us, e.g. we have observed the cloud-provider hang for 15 minutes without any actions just because of silent retries in SDK.  

@khenidak so, to achieve the final goal of large clusters, I think getting the retry-after back and let cloud-provider do the retry is better. WDYH?